code/game/objects/items/hand_items.dm

```diff
@@ -126,6 +128,28 @@
 	"<span class='hear'>You hear a slap.</span>")
 	return
 
+/obj/item/slapper/attack_obj(obj/hit_object, mob/living/user)
+	if(!istype(hit_object, /obj/structure/table))
+		return ..()
+
+	var/obj/structure/table/the_table = hit_object
+	user.do_attack_animation(the_table)
+	if(user.combat_mode && table_smacks_left == initial(table_smacks_left)) // so you can't do 2 weak slaps followed by a big slam
```

Comment:
Can you make this left-click/right-click? I want to avoid forcing people to constantly switch combat mode.

Reply:
I don't think `attack_obj_secondary` or whatever exists, but see if `attack_obj` passes params/modifiers at any point that you can just read.